pkg/planner/core/find_best_task.go

```diff
@@ -1356,6 +1356,10 @@ func findBestTask4DS(ds *DataSource, prop *property.PhysicalProperty, planCounte
 	for _, candidate := range candidates {
 		path := candidate.path
 		if path.PartialIndexPaths != nil {
+			// prefer tiflash, while current table path is tikv, skip it.
+			if ds.PreferStoreType&h.PreferTiFlash != 0 && path.StoreType == kv.TiKV {
```

Comment:
when we step into here, we can ensure that
* tiflash is existed (replica set is checked)
* whether tiflash path can be generated successfully or not, not sure here, just keep the same logic from L1498

future work: we can sort the candidate paths according to the user's preferred hints or variables, then gen them first, if we do, output their preferred path, on the condition that the preferred path can be generated successfully. IF failed, then try next, inferior paths